src/server/accessors/ExternalComponentsExtend.ts

```diff
@@ -0,0 +1,12 @@
+import { AppExternalComponentManager } from '../managers/AppExternalComponentManager';
+
+import { IExternalComponentsExtend } from '../../definition/accessors';
+import { IExternalComponent } from '../../definition/externalComponent/IExternalComponent';
+
+export class ExternalComponentsExtend implements IExternalComponentsExtend {
+    constructor(private readonly manager: AppExternalComponentManager, private readonly appId: string) { }
+
+    public async register(externalComponent: IExternalComponent): Promise<void> {
+        return Promise.resolve(this.manager.registerComponent(this.appId, externalComponent));
```

Comment:
You don't need to return a `Promise.resolve` here; as the method is `async`, it will return a `Promise` regardless of what it executes, even if the `registerComponent` method does not :)